platform/darwin/test/MGLExpressionTests.mm

```diff
@@ -300,6 +300,28 @@ - (void)testConstantValueExpressionObject {
         XCTAssertEqualObjects(expression.mgl_jsonExpressionObject, jsonExpression);
         XCTAssertEqualObjects([expression expressionValueWithObject:nil context:nil], color);
     }
+    {
+        // Transform color components to non-premultiplied values
+        float alpha = 0.5;
+        float red = (255.0 * alpha) / 255;
+        float green = (239.0 * alpha) / 255;
+        float blue = (213.0 * alpha) / 255;
+
+        MGLColor *color = [MGLColor mgl_colorWithColor:{ red, green, blue, alpha }]; // papayawhip
+        CGFloat originalRed = 0.0, originalGreen = 0.0, originalBlue = 0.0, originalAlpha = 0.0;
+        [color getRed:&originalRed green:&originalGreen blue:&originalBlue alpha:&originalAlpha];
+
+        NSArray *jsonExpression = @[@"rgba", @255.0, @239.0, @213.0, @0.5];
+        NSExpression *papayawhipExpression = [NSExpression expressionWithMGLJSONObject:jsonExpression];
+        MGLColor *papayaWhipColor = [papayawhipExpression expressionValueWithObject:nil context:nil];
+        CGFloat convertedRed = 0.0, convertedGreen = 0.0, convertedBlue = 0.0, convertedAlpha = 0.0;
+        [papayaWhipColor getRed:&convertedRed green:&convertedGreen blue:&convertedBlue alpha:&convertedAlpha];
+
+        XCTAssertEqualWithAccuracy(convertedRed, originalRed, 0.0000001);
+        XCTAssertEqualWithAccuracy(convertedGreen, originalGreen, 0.0000001);
+        XCTAssertEqualWithAccuracy(convertedBlue, originalBlue, 0.0000001);
```

Comment:
I think it would be easier to sight-verify that this test is correct if the assertions tested equality to the elements of `jsonExpression` instead of these “original” variables that have already been round-tripped:

```objc
NSArray *jsonExpression = @[@"rgba", @255.0, @239.0, @213.0, @0.5];
MGLColor *papayaWhipColor = [NSExpression expressionWithMGLJSONObject:jsonExpression].constantValue;
CGFloat convertedRed = 0.0, convertedGreen = 0.0, convertedBlue = 0.0, convertedAlpha = 0.0;
[papayaWhipColor getRed:&convertedRed green:&convertedGreen blue:&convertedBlue alpha:&convertedAlpha];
XCTAssertEqualWithAccuracy(convertedRed, 255, 0.0000001);
XCTAssertEqualWithAccuracy(convertedGreen, 239, 0.0000001);
XCTAssertEqualWithAccuracy(convertedBlue, 213, 0.0000001);
XCTAssertEqual(convertedAlpha, 0.5);
```

This shorter version still provides coverage of `+[MGLColor mgl_colorWithColor:]`, just not directly.

Reply:
Got it, thanks for the input! I've made the changes to the test!